PubNub/Data/Managers/PNSubscriber.m

```diff
@@ -1061,11 +1049,19 @@ - (void)stopRetryTimer {
 #pragma mark - Handlers
 
 - (void)handleSubscriptionStatus:(PNSubscribeStatus *)status {
-
+    
     [self stopRetryTimer];
     if (!status.isError && status.category != PNCancelledCategory) {
-        
         [self handleSuccessSubscriptionStatus:status];
+#ifdef PGDROID
+        // 2016/02/16 : this PG addition is causing crashes on Android ... Sending an instance of PNErrorStatus to -handleSuccessSubscriptionStatus: is not valid since it references properties specific to PNSubscribeStatus ... unsure why we're not seeing this in crash groups on iOS ...  LONG TERM solution is to probably revert both of the commits around this and roll forward to the latest PubNub and run full QA ;-)
+#else
+    } else if (status.isError && status.category == PNTimeoutCategory) {
+        // Treat timeouts as successful because there isn't a good way of knowing
+        // why we are disconnecting. This was cherry-picked from here:
+        // https://github.com/pocketgems/PubNub/commit/05eb350b448e12b746ea94b480abc753da162021
+        [self handleSuccessSubscriptionStatus:status];
```

Comment:
This seems like a bad decision made long time ago. Should we just leave it as is? @mjk3979

Reply:
I think we should not readd this code to dragons. There was a complicated reason for needing this change, but I think we changed the timeout back to a reasonable value, so this shouldn't be needed anymore

Reply:
Not knowing anything about it, I'd say there's a chance this will make things worse, so I'm okay with reverting this. I'll go ahead and revert the commits.